rfc/rfc-43/rfc-43.md

```diff
@@ -0,0 +1,369 @@
+
+With the increase in the number of HUDI tables, due to a lack of management capabilities, maintenance costs will become
+higher. This proposal is to implement an independent compaction/clustering Service to manage the Hudi
+compaction/clustering job.
+
+## Background
+
+In the current implementation, if the HUDI table needs do compact/cluster, it only has three ways:
+
+1. Use inline compaction/clustering, in this mode the job will be block writing job.
+
+2. Using Async compaction/clustering, in this mode the job execute async but also sharing the resource with HUDI to
+   write a job that may affect the stability of job writing, which is not what the user wants to see.
+
+3. Using independent compaction/clustering job is a better way to schedule the job, in this mode the job execute async
+   and do not sharing resources with writing job, but also has some questions:
+    1. Users have to enable lock service providers so that there is not data loss. Especially when compaction/clustering
+       is getting scheduled, no other writes should proceed concurrently and hence a lock is required.
+    2. The user needs to manually start an async compaction/clustering application, which means that the user needs to
+       maintain two jobs.
+    3. With the increase in the number of HUDI jobs, there is no unified service to manage compaction/clustering jobs (
+       monitor, retry, history, etc...), which will make maintenance costs increase.
+
+With this effort, we want to provide an independent compaction/clustering Service, it will have these abilities:
+
+- Provides a pluggable execution interface that can adapt to multiple execution engines, such as Spark and Flink.
+
+- With the ability to failover, need to be persisted compaction/clustering message.
+
+- Perfect metrics and reuse HoodieMetric expose to the outside.
+
+- Provide automatic failure retry for compaction/clustering job.
+
+## Implementation
+
+### Processing mode
+
+Different processing modes depending on whether the meta server is enabled
+
+- Hudi metaserver is used
+    - The pull-based mechanism works for fewer tables. Scanning 1000s of tables for possible services is going to induce
+      lots of a load of listing.
+    - The meta server provides a listener that takes as input the uris of the Table Management Service and triggers a
+      callback through the hook at each instant commit, thereby calling the Table Management Service to do the
+      scheduling/execution for the table.
+      ![](service_with_meta_server.png)
+
+- Hudi metaserver is not used
+    - for every write/commit on the table, the table management server is notified.
+    - Each request to the table management server carries all pending instant matches the current action type.
+      ![](service_without_meta_server.png)
+
+### Processing flow
+
+- If hudi metaserver is used, after receiving the request, the table management server schedules the relevant table
+  service to the table's timeline
```

Comment:
> schedules the relevant table service to the table's timeline

need to make it explicit: this is table timeline managed in metaserver, right? it can confuse with the table timeline on storage. Should also mention how metaserver interact with storage in this case.